libsnark/gadgetlib1/gadgets/hashes/anemoi/tests/test_anemoi_gadget.cpp

```diff
@@ -360,10 +366,72 @@ template<typename ppT> void test_for_curve()
     test_anemoi_permutation_mds<ppT, parameters>();
 }
 
+template<typename ppT, class parameters = anemoi_parameters<libff::Fr<ppT>>>
+void test_curve_parameters()
+{
+    printf("g = %zd\n", parameters::multiplicative_generator_g);
+    printf("alpha = %zd\n", parameters::alpha);
+    printf("beta = %zd\n", parameters::beta);
+    printf("gamma = %zd\n", parameters::gamma);
+    printf("quad_exponent = %zd\n", parameters::quad_exponent);
+    printf("alpha_inv = ");
+    parameters::alpha_inv.print();
+    printf("delta = ");
+    parameters::delta.print();
+}
+
 TEST(TestAnemoiGadget, BLS12_381) { test_for_curve<libff::bls12_381_pp>(); }
 
+TEST(TestCurveParameters, BLS12_381)
+{
+    using parameters = anemoi_parameters<libff::bls12_381_pp>;
+    test_curve_parameters<libff::bls12_381_pp, parameters>();
+}
+
+TEST(TestCurveParameters, BLS12_377)
+{
+    using parameters = anemoi_parameters<libff::bls12_377_pp>;
```

Comment:
Are all the other tests specific to bls12-381?

Even though we don't have any test vectors, it would be nice to instantiate the gadgets.  This allows the compiler to check for any inconsistencies.

If the other test functions *are* specific to bls12-381, they should not be parameterised by the curve.  But ideally we'd be able to do some basic checks that the gadgets create proofs and witnesses.